### PR TITLE
Add support new authentication types and request proxy for Synthetics tests

### PR DESCRIPTION
**Pull request by romainberger**
Add new authentication types for API tests and request proxy for browser and api tests.

### CODE REVIEW

**Review by AlaricCalmette — DISMISSED** [review]

File: docs/resources/synthetics_test.md [hunk 1]
```diff
@@ -380,10 +382,18 @@ Optional:
 <a id="nestedblock--api_step--request_basicauth"></a>
 ### Nested Schema for `api_step.request_basicauth`
 
-Required:
+Optional:
 
+- **access_key** (String) Access key for the `SIGV4` authentication.
+- **domain** (String) Domain for the `ntlm` authentication.
```
Comment (maycmlee) [inline docs/resources/synthetics_test.md:388]:
```suggestion
- **access_key** (String) Access key for `SIGV4` authentication.
- **domain** (String) Domain for `ntlm` authentication.
```

File: docs/resources/synthetics_test.md [hunk 2]
```diff
@@ -380,10 +382,18 @@ Optional:
...
 - **password** (String, Sensitive) Password for authentication.
+- **region** (String) Region for the `SIGV4` authentication.
```
Comment (maycmlee) [inline docs/resources/synthetics_test.md:390]:
```suggestion
- **region** (String) Region for `SIGV4` authentication.
```

File: docs/resources/synthetics_test.md [hunk 3]
```diff
@@ -380,10 +382,18 @@ Optional:
...
+- **secret_key** (String) Secret key for the `SIGV4` authentication.
```
Comment (maycmlee) [inline docs/resources/synthetics_test.md:391]:
```suggestion
- **secret_key** (String) Secret key for `SIGV4` authentication.
```

File: docs/resources/synthetics_test.md [hunk 4]
```diff
@@ -380,10 +382,18 @@ Optional:
...
+- **service_name** (String) Service name for the `SIGV4` authentication.
```
Comment (maycmlee) [inline docs/resources/synthetics_test.md:392]:
```suggestion
- **service_name** (String) Service name for `SIGV4` authentication.
```

File: docs/resources/synthetics_test.md [hunk 5]
```diff
@@ -380,10 +382,18 @@ Optional:
...
+- **session_token** (String) Session token for the `SIGV4` authentication.
```
Comment (maycmlee) [inline docs/resources/synthetics_test.md:393]:
```suggestion
- **session_token** (String) Session token for `SIGV4` authentication.
```

File: docs/resources/synthetics_test.md [hunk 6]
```diff
@@ -380,10 +382,18 @@ Optional:
...
+- **type** (String) The type of basic authentication to use when performing the test.
```
Comment (maycmlee) [inline docs/resources/synthetics_test.md:394]:
```suggestion
- **type** (String) Type of basic authentication to use when performing the test.
```

File: docs/resources/synthetics_test.md [hunk 7]
```diff
@@ -380,10 +382,18 @@ Optional:
...
 - **username** (String) Username for authentication.
+- **workstation** (String) workstation for the `ntlm` authentication.
```
Comment (maycmlee) [inline docs/resources/synthetics_test.md:396]:
```suggestion
- **workstation** (String) Workstation for `ntlm` authentication.
```

File: docs/resources/synthetics_test.md [hunk 8]
```diff
@@ -597,10 +619,18 @@ Optional:
 <a id="nestedblock--request_basicauth"></a>
 ### Nested Schema for `request_basicauth`
 
-Required:
+Optional:
 
+- **access_key** (String) Access key for the `SIGV4` authentication.
```
Comment (maycmlee) [inline docs/resources/synthetics_test.md:624]: Same as above comment

File: datadog/resource_datadog_synthetics_test_.go [hunk 9]
```diff
@@ -1322,6 +1406,23 @@ func completeSyntheticsTestRequest(request datadogV1.SyntheticsTestRequest, requ
 		request.SetCertificate(requestClientCertificate)
 	}
 
+	if len(requestProxy) > 0 {
+		if proxy, ok := requestProxy[0].(map[string]interface{}); ok {
+			requestProxy := datadogV1.SyntheticsTestRequestProxy{}
```
Comment (skarimo) [inline datadog/resource_datadog_synthetics_test_.go:1411]: Could we rename the `requestProxy` variable here so it doesn't clash with the one from outside scope

File: datadog/tests/resource_datadog_synthetics_test_test.go [hunk 10]
```diff
@@ -2291,6 +2370,11 @@ resource "datadog_synthetics_test" "bar" {
                url = "https://www.datadoghq.com"
        }
 
+       request_basicauth {
+		       username = "ntlm-username"
+		       password = "ntlm-password"
```
Comment (skarimo) [inline datadog/tests/resource_datadog_synthetics_test_test.go:2375]:
```suggestion
		       username = "web-username"
		       password = "web-password"
```
Small nit

File: datadog/tests/resource_datadog_synthetics_test_test.go [hunk 11]
```diff
@@ -2229,6 +2300,14 @@ func createSyntheticsBrowserTestBrowserVariablesStep(ctx context.Context, accPro
...
 			resource.TestCheckResourceAttr(
 				"datadog_synthetics_test.bar", "request_definition.0.url", "https://www.datadoghq.com"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.bar", "request_basicauth.#", "1"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.bar", "request_basicauth.0.type", "web"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.bar", "request_basicauth.0.username", "ntlm-username"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.bar", "request_basicauth.0.password", "ntlm-password"),
```
Comment (skarimo) [inline datadog/tests/resource_datadog_synthetics_test_test.go:2310]:
```suggestion
				"datadog_synthetics_test.bar", "request_basicauth.0.username", "web-username"),
			resource.TestCheckResourceAttr(
				"datadog_synthetics_test.bar", "request_basicauth.0.password", "web-password"),
```

File: datadog/resource_datadog_synthetics_test_.go [hunk 12]
```diff
@@ -222,17 +223,78 @@ func syntheticsTestRequestBasicAuth() *schema.Schema {
...
+					Optional:    true,
 					Sensitive:   true,
 				},
+				"access_key": {
+					Type:        schema.TypeString,
+					Description: "Access key for `SIGV4` authentication.",
+					Optional:    true,
+				},
+				"secret_key": {
+					Type:        schema.TypeString,
```
Comment (skarimo) [inline datadog/resource_datadog_synthetics_test_.go:250]: We should mark `secret_key` and `access_key` as sensitive

**Review by skarimo — COMMENTED** [review]
👋 Left a few nits but otherwise LGTM. Could you run `make docs`. Thanks

**Pull request merged**